plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -174,6 +175,12 @@ func NodeRules() []rbacv1.PolicyRule {
 
 	// RuntimeClass
 	nodePolicyRules = append(nodePolicyRules, rbacv1helpers.NewRule("get", "list", "watch").Groups("node.k8s.io").Resources("runtimeclasses").RuleOrDie())
+
+	// DRA Resource Claims
+	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
+		nodePolicyRules = append(nodePolicyRules, rbacv1helpers.NewRule("get").Groups(resourceGroup).Resources("resourceclaims").RuleOrDie())
```

Comment:
The node should be able to read a ResourceClaim if any pod scheduled onto the node references the ResourceClaim. That is visible in the API, so this should be doable.

I have created https://github.com/kubernetes/kubernetes/issues/113866 to track this.

Reply:
> The node should be able to read a ResourceClaim if any pod scheduled onto the node references the ResourceClaim. That is visible in the API (pod.spec.resourceClaims), so this should be doable.

That makes sense for pods that statically reference resource claims. What about pods that reference `resourceClaimTemplateName` and claims instances are dynamically generated?

Reply:
oh, I just saw the API doc:

> The name of the ResourceClaim will be <pod name>-<resource name>, where <resource name> is the PodResourceClaim.Name.

That works. (also, I hope we're preventing `-` in PodResourceClaim.Name)

Reply:
Right now, the name of those generated ResourceClaims is derived from pod name and the pod.spec.resourceClaims[].name, i.e. deterministic. So the same logic for a ResourceClaim that is named explictly works. This is the same approach that was used for generic ephemeral volumes, exactly because code which needs a name can compute it without having to search. The downside is the risk of conflicts with existing objects, but those are handled and don't seem to be a problem in practice.

For ResourceClaim we don't have a lot of existing code, so we may still decide to generate ResourceClaims with a generated name. This is tracked in https://github.com/kubernetes/kubernetes/issues/113722. We will have to make sure that this then still works also for the node authorizer.

Reply:
> also, I hope we're preventing - in PodResourceClaim.Name

looks like we aren't. we should, otherwise that construction is ambiguous (pod `a` with resourcename `b-c` and pod `a-b` with resourcename `c` collide)

Reply:
That is true. It's left to users to avoid that, but not allowing the problem to occur would be better,

I've added a remark to https://github.com/kubernetes/kubernetes/issues/113722 that we either need to generate names or tighten validation.

Reply:
Tightening validation is not trivial https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#backward-compatibility-gotchas

Better do it now.

Reply:
I'd tend to agree… if this says this is how we construct this name in 1.26, it will never be easier to fix the collision than it is right now by forbidding "-"